snapcraft/internal/meta.py

```diff
@@ -253,33 +254,41 @@ def _write_wrap_exe(self, wrapexec, wrappath,
         cwd = 'cd {}'.format(cwd) if cwd else ''
 
         # If we are dealing with classic confinement it means all our
-        # binaries are linked with `nodefaultlib` so this is harmless.
-        # We do however want to be on the safe side and make sure no
-        # ABI breakage happens by accidentally loading a library from
-        # the classic system.
-        classic_library_paths = self._config_data['confinement'] == 'classic'
-        assembled_env = common.assemble_env(classic_library_paths,
-                                            self._arch_triplet)
-        assembled_env = assembled_env.replace(self._snap_dir, '$SNAP')
+        # binaries are linked with `nodefaultlib` but we still do
+        # not want to leak PATH or other environment variables
+        # that would affect the applications view of the classic
+        # environment it is dropped into.
         replace_path = r'{}/[a-z0-9][a-z0-9+-]*/install'.format(
             self._parts_dir)
```

Comment:
has `_parts_dir` been sanitized to not have regexpish things in it? otherwise you want `re.escape(self._parts_dir)` here